src/hash-join/concurrent_unordered_multimap.cuh

```diff
@@ -0,0 +1,515 @@
+/*
+ * Copyright (c) 2017-2018, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CONCURRENT_UNORDERED_MULTIMAP_CUH
+#define CONCURRENT_UNORDERED_MULTIMAP_CUH
+
+#include <iterator>
+#include <type_traits>
+#include <cassert>
+
+#include <thrust/pair.h>
+
+#include "managed_allocator.cuh"
+#include "managed.cuh"
+#include "hash_functions.cuh"
+
+// TODO: replace this with CUDA_TRY and propagate the error
```

Comment:
This TODO item is much needed.  Currently, CUDA_RT_CALL only prints the error message and keeps going, even though the error is likely unrecoverable.

Reply:
I added an exit after we hit an error in bcfcb27